servicetalk-transport-netty-internal/src/test/java/io/servicetalk/transport/netty/internal/SslCloseNotifyAlertClientHandlingTest.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright © 2020 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.transport.netty.internal;
+
+import io.servicetalk.concurrent.PublisherSource;
+import io.servicetalk.concurrent.api.test.StepVerifiers;
+
+import org.junit.Test;
+
+import java.nio.channels.ClosedChannelException;
+
+import static io.servicetalk.concurrent.api.Processors.newPublisherProcessor;
+import static io.servicetalk.concurrent.api.SourceAdapters.fromSource;
+
+public class SslCloseNotifyAlertClientHandlingTest extends AbstractSslCloseNotifyAlertHandlingTest {
```

Comment:
can you add a test case that verifies a Retryable exception is delivered if the first write fails due to prior `close_notify` being received?

Reply:
I've enhanced existing tests to expect a correct subtype of `ChannelClosedException` where it's appropriate. It covers the suggested use-case.